### DISCUSSION TITLE
Support for multiple having clauses

### DISCUSSION THREAD

**Discussion by liamw1** [diesel-rs/diesel#4117] [Ideas]
Hi, I'm using Diesel to construct some pretty complex queries. I need to be able to construct queries with dynamic numbers of filters, columns to sort on, and having clauses. I discovered that all query building operations except for filters will overwrite each other. For sorts I was able to get around this limitation with .then_order_by, but there doesn't seem to be an equivalent function to dynamically build up a HAVING clause.

Is there already a way to do something like this? If not, this feature would be very useful as having only one condition in my HAVING clause is a major limitation for me.

**Comment by weiznich** [comment]
This is currently not supported, but we are open to receive PR's adding support for this in a similar manner than this was done for `filter` and `order`. Such a PR should contain some documentation and some tests.
